transactions/src/main/java/org/ehcache/transactions/xa/internal/XAStore.java

```diff
@@ -792,15 +794,16 @@ public Duration getExpiryForCreation(K key, SoftLock softLock) {
         }
 
         @Override
-        public Duration getExpiryForAccess(K key, SoftLock softLock) {
-          if (softLock.getTransactionId() != null) {
+        public Duration getExpiryForAccess(K key, final ValueSupplier<? extends SoftLock<V>> softLock) {
+          if (softLock.value().getTransactionId() != null) {
             // phase 1 prepare, access -> forever
             return Duration.FOREVER;
           } else {
             // phase 2 commit, or during a TX's lifetime, access -> some time
             Duration duration;
             try {
-              duration = configuredExpiry.getExpiryForAccess(key, (V) softLock.getOldValue());
+              ValueSupplier<V> supplierExpired = supplierOf(softLock.value().getOldValue());
```

Comment:
Did we want to delay the softLock.value() resolution too?  Is there any advantage to that @lorban ?